src/index.js

```diff
@@ -30,11 +30,30 @@ class Container extends Component {
   }
 
   prepareInitialStore = (options = {}, prevOptions = {}) => {
-    const { userDetails: { smsNumber } = {} } = options
-    const { userDetails: { smsNumber: prevSmsNumber } = {} } = prevOptions
+    const { userDetails: { smsNumber } = {}, steps } = options
+    const { userDetails: { smsNumber: prevSmsNumber } = {}, steps: prevSteps } = prevOptions
 
-    if (smsNumber && smsNumber !== prevSmsNumber)
+    if (smsNumber && smsNumber !== prevSmsNumber) {
       actions.setMobileNumber(smsNumber)
+    }
+
+    if (steps && steps !== prevSteps) {
+      this.handledPreselectedDocument(steps, actions)
+    }
+  }
+
+  handledPreselectedDocument = (steps, actions) => {
+    const isDocument = (step) => step.type === 'document'
+    const documentStep = Array.find(steps, isDocument)
+    const hasDocTypes = documentStep && documentStep.options && documentStep.options.documentTypes
+    if (hasDocTypes) {
+      const docTypes = documentStep.options.documentTypes
+      const enabledDocTypes = Object.keys(docTypes).filter((type) => docTypes[type])
+      if (enabledDocTypes.length === 1) {
+        actions.setDocumentType(enabledDocTypes[0])
+        actions.hasPreselectedDoc(true)
```

Comment:
why is this going to store? It's not really state that will be modified internally.